### ISSUE TITLE
Activity opens to a black canvas

### ISSUE THREAD

**Issue by chimosky** [sugarlabs/falabracman-activity#20]
Noticed this while testing on Sugar 0.121, pygame 2.1.2 (SDL 2.26.5, Python 3.11.2).

I'm guessing it might be an issue with pygame like it was last time, not sure though as I 
haven't taken a look.

This was in the log of another activity, seems it's happening to pygame activities.
```python3
(sugar-activity3:1932): Gtk-CRITICAL **: 12:01:23.603: gtk_widget_draw: assertion '!widget->priv->alloc_needed' failed
```

**Comment by quozl** [comment]
Could it be issue 14 of sugargame?  If so, not worth tracking on a per-activity basis.  :grin:

**Comment by chimosky** [comment]
> Could it be issue 14 of sugargame? If so, not worth tracking on a per-activity basis. 😁

Agreed, couldn't look at the test activity at the time so I just opened here, although it doesn't seem to be an issue with sugargame as the test activity opens and not to a blank canvas, would open an issue there for something I noticed and have seen in some other activities recently.

**Comment by quozl** [comment]
So it will depend on the SDL version (for when SDL_WINDOWID was removed), and then the Pygame version (for when SDL_WINDOWID was added back).

**Comment by chimosky** [comment]
> So it will depend on the SDL version (for when SDL_WINDOWID was removed), and then the Pygame version (for when SDL_WINDOWID was added back).

Yes, but I updated a while back and the issue went away, this just happened again and if that turns out to be the problem then why add it back just to remove it?

**Comment by quozl** [comment]
> Yes, but I updated a while back and the issue went away, this just happened again and if that turns out to be the problem then why add it back just to remove it?

Different open source projects are involved; one project removed the feature, another project was relying upon it, and so reimplemented it again.  You will find SDL_WINDOWID resolved inside Pygame now, and yet why isn't it named PYGAME_WINDOWID?  Because many people had written Pygame apps and used SDL_WINDOWID.

In the long run, it would have been better had Pygame project years ago responded to the need to wrap their windows by creating a Pygame feature that, underneath, used SDL in whatever way was required.
